halo2-ecc/src/ecc/schnorr_signature.rs

```diff
@@ -0,0 +1,83 @@
+use crate::bigint::{big_is_equal, ProperCrtUint};
+use crate::fields::{fp::FpChip, FieldChip, PrimeField};
+use halo2_base::{gates::GateInstructions, utils::CurveAffineExt, AssignedValue, Context};
+
+use super::{fixed_base, scalar_multiply, EcPoint, EccChip};
+
+// CF is the coordinate field of GA
+// SF is the scalar field of GA
+// p = base field modulus
+// n = scalar field modulus
+/// `pubkey` should not be the identity point
+/// follow spec in https://github.com/bitcoin/bips/blob/master/bip-0340.mediawiki
+pub fn schnorr_verify_no_pubkey_check<F: PrimeField, CF: PrimeField, SF: PrimeField, GA>(
+    chip: &EccChip<F, FpChip<F, CF>>,
+    ctx: &mut Context<F>,
+    pubkey: EcPoint<F, <FpChip<F, CF> as FieldChip<F>>::FieldPoint>,
+    r: ProperCrtUint<F>,       // int(sig[0:32]); fail if r ≥ p.
+    s: ProperCrtUint<F>,       // int(sig[32:64]); fail if s ≥ n
+    msgHash: ProperCrtUint<F>, // int(hashBIP0340/challenge(bytes(r) || bytes(P) || m)) mod n
+    var_window_bits: usize,
+    fixed_window_bits: usize,
+) -> AssignedValue<F>
+where
+    GA: CurveAffineExt<Base = CF, ScalarExt = SF>,
+{
+    let base_chip = chip.field_chip;
+    let scalar_chip =
+        FpChip::<F, SF>::new(base_chip.range, base_chip.limb_bits, base_chip.num_limbs);
+
+    // check r < p
+    let r_valid = base_chip.is_soft_nonzero(ctx, &r);
```

Comment:
This prevents `r = 0`. For most curves there's no way `x(R) = 0` so it seems technically ok, but wanted to check if that's the intention

Reply:
updated to not prevent r = 0